docs/core/extensions/logging-library-authors.md

```diff
@@ -0,0 +1,85 @@
+
+Instead, you should accept an <xref:Microsoft.Extensions.Logging.ILoggerFactory> and use it to create <xref:Microsoft.Extensions.Logging.ILogger%601> instances. Using an `ILoggerFactory` ensures that the logging configuration is consistent with the rest of the app and is more flexible. For more information, see [Log category](logging.md#log-category).
+
+Logging supports filtering only by log level and category, so it's important to use the correct category for each log message. For more information, see [How filtering rules are applied](logging.md#how-filtering-rules-are-applied).
+
+### Create a logger factory
+
+To create an `ILoggerFactory`, use the <xref:Microsoft.Extensions.Logging.LoggerFactory> class. This class is available from the [Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging) NuGet package. The following example shows how to create a logger factory that logs to the console:
+
+```csharp
+var factory = LoggerFactory.Create(builder =>
+{
+    builder.AddConsole();
+});
+```
+
+> [!TIP]
+> The <xref:Microsoft.Extensions.Logging.ConsoleLoggerExtensions.AddConsole%2A> extension method is available from the [Microsoft.Extensions.Logging.Console](https://www.nuget.org/packages/Microsoft.Extensions.Logging.Console) NuGet package. For more information, see [Logging providers in .NET](logging-providers.md).
+
+### Default with null logging
+
+Libraries can default to _null logging_ if no `ILoggerFactory` is provided. The use of _null logging_ differs from defining types as nullable (`ILoggerFactory?`), as the types are non-null. These convenience-based types don't log anything and are essentially no-ops. Consider using any of the available abstraction types:
+
+- <xref:Microsoft.Extensions.Logging.Abstractions.NullLogger>
+- <xref:Microsoft.Extensions.Logging.Abstractions.NullLoggerFactory>
+- <xref:Microsoft.Extensions.Logging.Abstractions.NullLoggerProvider>
+
+### Dependency injection-aware libraries
+
+If your library is dependency injection-aware, classes that require logging may accept an <xref:Microsoft.Extensions.Logging.ILogger%601> in the constructor.
+
+:::code source="./snippets/logging/library-authors/DiExampleService.cs":::
+
+For more information, see [.NET dependency injection](dependency-injection.md).
+
+### Dependency injection-oblivious libraries
+
+If your library is dependency injection-oblivious, classes that require logging should use the configured <xref:Microsoft.Extensions.Logging.ILoggerFactory> to resolve <xref:Microsoft.Extensions.Logging.ILogger%601> instances on demand.
```

Comment:
I've never heard it phrased as 'dependency injection-oblivious'. Is this a common way we say it in docs? I typically say something like "If your library isn't DI-aware" or "If your library isn't designed for DI"

Reply:
Yeah, I'm not sure if it's very common either - but docs are often a place where terms are coined. DI-aware and DI-oblivious are antonyms. But I agree, reading it again it's a bit clunky. I'll reword it.